google/resource_container_cluster.go

```diff
@@ -364,10 +367,19 @@ func resourceContainerCluster() *schema.Resource {
 			},
 
 			"enable_shielded_nodes": {
+				Type:          schema.TypeBool,
+				Optional:      true,
+				Default:       true,
```

Comment:
@c2thorn Enabling Shielded Nodes by default is in alignment with GKE 1.18.

I'm seeing test failures (for example `TestAccContainerCluster_withMasterAuthConfig`) when shielded nodes is `false`:

```
# google_container_cluster.with_master_auth_no_cert will be updated in-place
          ~ resource "google_container_cluster" "with_master_auth_no_cert" {
              ~ enable_shielded_nodes       = true -> false
```